sim/dex-data.ts

```diff
@@ -905,70 +911,72 @@ export class Move extends BasicEffect implements Readonly<BasicEffect & MoveData
 		this.stab = data.stab || undefined;
 		this.volatileStatus = typeof data.volatileStatus === 'string' ? (data.volatileStatus as ID) : undefined;
 
-		if (this.category !== 'Status' && !this.gmaxPower) {
+		if (this.category !== 'Status' && !this.maxMove?.basePower) {
+			this.maxMove = this.maxMove || {} as MoveData['maxMove'];
```

Comment:
```suggestion
			this.maxMove = {};
```

With the `!this.maxMove` change, there's no need for this to be this complicated.

Reply:
This cast is still needed because `basePower` is not optional.